tests/jballerina-unit-test/src/test/resources/test-src/klass/service_type_resource_method_decl_neg.bal

```diff
@@ -23,3 +23,23 @@ type SType service object {
 service class SClass {
     *SType;
 }
+
+service class ROne {
+    resource function get foo() returns string {
+        return "foo";
+    }
+}
+
+// Not having a implementation for "get foo" resource method is expected as resource method is not part of the obj type.
+service class RTwo {
```

Comment:
Shall we also add an assignability tests with inclusions?

```ballerina
service class Foo {
    resource function get greeting() returns string => "hello";
}

service class Bar {
    *Foo;

    function fn() {

    }
}

service class Baz {
    function fn() {
        
    }
}

public function main() {
    Bar x = new Baz(); // allowed now
}
```

Reply:
Came across https://github.com/ballerina-platform/ballerina-lang/issues/31906 btw.

Reply:
Added with https://github.com/ballerina-platform/ballerina-lang/pull/31761/commits/ada5ceb039c3ea0e94ad9b091ceab82b76df7a64